### PR TITLE
Fix tests to use Ahem, and helpful changes around that

### PR DESCRIPTION
**Pull request by Hixie**
- Fix fonts that had metric-specific behaviours.

- LiveTestWidgetsFlutterBinding.allowAllFrames has been renamed
  to LiveTestWidgetsFlutterBinding.framePolicy.

- LiveTestWidgetsFlutterBinding now defaults to using a frame policy
  that pumps slightly more frames, to animate the pointer crosshairs.

- Added "flutter run --use-test-fonts" to enable Ahem on devices.

- Changed how idle() works to be more effective in live mode.

- Display the test name in live mode (unless ahem fonts are enabled).

- Added a toString to TextSelectionPoint.

- Style nit fixes.

### CODE REVIEW

File: packages/flutter/test/material/text_field_test.dart [hunk 1]
```diff
@@ -41,12 +40,12 @@ void main() {
   SystemChannels.platform.setMockMethodCallHandler(mockClipboard.handleMethodCall);
 
   const String kThreeLines =
-    'First line of text is here abcdef ghijkl mnopqrst. ' +
-    'Second line of text goes until abcdef ghijkl mnopq. ' +
-    'Third line of stuff keeps going until abcdef ghijk. ';
+    'First line of text is ' +
```
Comment (HansMuller) [inline packages/flutter/test/material/text_field_test.dart]: I assume that all of the abcd... stuff was to force the text field to wrap. Is that not needed anymore?

File: packages/flutter_test/lib/src/binding.dart [hunk 2]
```diff
@@ -857,16 +938,37 @@ class _LiveTestPointerRecord {
...
+    fontSize: 10.0,
+  );
+  void _setDescription(String value) {
+    assert(value != null);
+    if (value.isEmpty) {
+      _label = null;
+      return;
+    }
+    _label ??= new TextPainter(textAlign: TextAlign.left);
+    _label.text = new TextSpan(text: value, style: _labelStyle);
```
Comment (HansMuller) [inline packages/flutter_test/lib/src/binding.dart]: I suppose you could use ..text and ..layout here ...

File: packages/flutter/test/material/text_field_test.dart (hunk 1 above)
Comment (Hixie) [inline packages/flutter/test/material/text_field_test.dart]: Yeah the Ahem font is much wider than most fonts.

**Comment by HansMuller** [comment]
I've looked through everything. I can't say I understood the implications of each change, but everything looks sensible, and some things just look better.

LGTM

File: packages/flutter_test/lib/src/binding.dart (hunk 2 above)
Comment (Hixie) [inline packages/flutter_test/lib/src/binding.dart]: Does that work right with `??=`? I want to set the text property even when `_label` is already constructed.

Comment (HansMuller) [inline packages/flutter_test/lib/src/binding.dart]: Good point. The entire RHS only executes when the LHS is null, which is not what you wanted.

**Pull request merged**